[[ Perf ]] Improve get speed

runrevmark opened the pull request:
The setting of it has been sped up by not indirecting through an MCContainer.

The speed of 'get' has been improved by vectoring through an MCExecValue rather than an MCValueRef.

runrevmark committed:
engine: Improve performance of setting 'it'.

The SetValueToIt method was unnecessarily indirecting through an
MCContainer.

Using an MCContainer isn't necessary in this context, because there
is never any array path.

runrevmark committed:
engine: Change 'get' to use ExecValues.

The get command now uses an MCExecValue to collect the value of
the expression to set 'it' to. This cuts out ValueRef construction
for types not needing one (e.g. numbers).

livecodefraser commented:
@livecode-vulcan review okay b659de1

livecode-vulcan commented:
💙 review by @livecodefraser ok b659de167881223110f0bd73947920c9d931c2d6

peter-b commented:
@livecode-vulcan review not ok b659de1

Per conversation with @runrevmark this should use a `GiveValueToIt()` method (or some equivalent) to facilitate grepability.

livecode-vulcan commented:
💔 review by @peter-b not ok b659de167881223110f0bd73947920c9d931c2d6

runrevmark committed:
engine: Added GiveValueToIt method to MCExecContext.

In order to ensure we can easily find where It is mutated by syntax,
direct access to it in MCEngineExecGet has been replaced by a wrapper
GiveValueToIt.

livecodeali commented:
Would it be worth doing this with MCresult too to speed up returning a number?

peter-b commented:
@livecode-vulcan review okay 56b69c4

livecode-vulcan commented:
💙 review by @peter-b ok 56b69c4c675c23aaa0dc0c33611f52e53ec4719f

livecode-vulcan commented:
😎 test success 56b69c4c675c23aaa0dc0c33611f52e53ec4719f
- try-community-linux-x86: [success](https://vulcan.livecode.com/builders/try-community-linux-x86/builds/611)
- try-community-mac: [success](https://vulcan.livecode.com/builders/try-community-mac/builds/623)
- try-community-linux-x86_64: [success](https://vulcan.livecode.com/builders/try-community-linux-x86_64/builds/612)
- try-community-emscripten: [success](https://vulcan.livecode.com/builders/try-community-emscripten/builds/613)
- try-community-ios: [success](https://vulcan.livecode.com/builders/try-community-ios/builds/622)
- try-community-win-x86: [success](https://vulcan.livecode.com/builders/try-community-win-x86/builds/633)
- try-community-android-armv6: [success](https://vulcan.livecode.com/builders/try-community-android-armv6/builds/624)

livecodesebastien commented:
Should we backport it to 7.0?
